mas-foundation/src/main/java/com/ca/mas/core/storage/sharedstorage/AccountManagerUtil.java

```diff
@@ -51,27 +53,35 @@ public AccountManagerUtil(Context context, String accountName, boolean sharedSto
             mAccountManager = AccountManager.get(MAS.getContext());
             //Attempt to retrieve the account
             Account[] accounts = mAccountManager.getAccountsByType(accountType);
+            sb.append("No of accounts:: "+accounts.length);
             for (Account account : accounts) {
-                if (accountName.equals(account.name)) {
+                if (accountName != null &&accountName.equals(account.name)) {
                     String password = mAccountManager.getPassword(account);
                     String savedPassword = identifier.toString();
-                    if (password.equals(savedPassword)) {
+                    if (password != null && password.equals(savedPassword)) {
                         mAccount = account;
                     }else {
                         // - case migration from old AccountManagerStoreDataSource
                         mAccount = null;
                         identifier = new SharedStorageIdentifier();
                     }
+                } else {
```

Comment:
We might be throwing the exception for name mismatch. It's a mistake.

We need to rewrite the logic like below:

accountName is invariant to the loop. Hence, a null check for the **_accountName_** should be done outside the for-loop.

```
if (acoountName == null) {
throw new IllegalArgumentException("Invalid parameters, Account name cannot be null");
}

for (Account account : accounts) {
     if (accountName.equals(account.name)) {
     }
}
```